src/rust/src/backend/ec.rs

```diff
@@ -274,14 +274,35 @@ impl ECPrivateKey {
             ));
         }
 
-        let (data, _) = utils::calculate_digest_and_algorithm(
+        let (data, _algo) = utils::calculate_digest_and_algorithm(
             py,
             data.as_bytes(),
             signature_algorithm.getattr(pyo3::intern!(py, "algorithm"))?,
         )?;
 
         let mut signer = openssl::pkey_ctx::PkeyCtx::new(&self.pkey)?;
         signer.sign_init()?;
+        let deterministic: bool = signature_algorithm
+            .getattr(pyo3::intern!(py, "deterministic_signing"))?
+            .extract()?;
+        cfg_if::cfg_if! {
+            if #[cfg(CRYPTOGRAPHY_OPENSSL_320_OR_GREATER)]{
+                match deterministic {
+                    true => {
+                        let hash_function_name = _algo.getattr(pyo3::intern!(py, "name"))?.extract::<&str>()?;
+                        let hash_function = openssl::md::Md::fetch(None, hash_function_name, None)?;
```

Comment:
There's a function in the hashes module that you can use for this.

Reply:
I tried, but I'm having trouble constructing the appropriate type. The `hashes` function I tried to use returns a `openssl::hash::MessageDigest`:
```rust
pub(crate) fn message_digest_from_algorithm(
    py: pyo3::Python<'_>,
    algorithm: &pyo3::PyAny,
) -> CryptographyResult<openssl::hash::MessageDigest> {
```
The function I'm calling that uses the hash function takes an `openssl::md::MdRef`:
```rust
pub fn set_signature_md(&self, md: &MdRef) -> Result<(), ErrorStack> 
```

I can convert one to the other by adding unsafe code:
```rust
let hash_function = hashes::message_digest_from_algorithm(py, _algo)?;
unsafe {
    signer.set_signature_md(&openssl::md::Md::from_ptr(hash_function.as_ptr().cast_mut()))?;
}
 ```
 But not sure if that's what we want (as opposed to directly using openssl::md::Md::fetch(), which already encapsulates the unsafe use)